localstack-core/localstack/services/apigateway/helpers.py

```diff
@@ -1503,12 +1503,12 @@ def get_event_request_context(invocation_context: ApiInvocationContext):
     resource_id = invocation_context.resource_id
 
     set_api_id_stage_invocation_path(invocation_context)
-    relative_path, query_string_params = extract_query_string_params(
-        path=invocation_context.path_with_query_string
-    )
     api_id = invocation_context.api_id
     stage = invocation_context.stage
 
+    full_path = invocation_context.raw_uri.removeprefix(f"/{stage}")
+    relative_path, query_string_params = extract_query_string_params(path=full_path)
+
```

Comment:
This is probably where the failing test comes from. As the `raw_uri` contains `_user_request_`. That seems like a pretty bad naming for `path_with_query_string`, to also be responsible for removing the `_user_request_`... But we will be moving away from it soon! :rofl: